### ISSUE TITLE
Implement Cache System

### ISSUE THREAD

**Issue by jdauphant** [snipsco/snipskit-swift#1]
- [x] Max size (LRU)
- [ ] Expiration Date
- [ ] Persistance

**Comment by maelp** [comment]
BTW are you storing this in database or using NSCache? also if you need help with all geometrical methods (search locations in a radius, compute distance to location) just ask :)

**Comment by jeremiegirault** [comment]
At the moment, pure swift implementation.
We thought about a great system on friday to handle geometrical issues, maybe you will be able to help for the geometric cache implementation. 


-- 
Jérémie Girault
Software Engineer - snips.net
jeremie.girault@snips.net
Mobile: +33 6 81 71 18 16
Skype: jeremie.girault
Twitter: @kamidude
LinkedIn: linkedin.com/in/jeremiegirault

VSRE compliant - vsre.info


On 24 août 2015 at 11:49:53, Mael (notifications@github.com) wrote:

BTW are you storing this in database or using NSCache? also if you need help with all geometrical methods (search locations in a radius, compute distance to location) just ask :)

**Comment by maelp** [comment]
We can put things in the database — actually there is already the PlaceCache implementation in the SDK which does almost exactly what you need

**Comment by jeremiegirault** [comment]
Yes, but we need it decoupled from the SDK (remember this issue tracker is public, thanks).
Persistance will be addable by creating a custom cache backend.


-- 
Jérémie Girault
Software Engineer - snips.net
jeremie.girault@snips.net
Mobile: +33 6 81 71 18 16
Skype: jeremie.girault
Twitter: @kamidude
LinkedIn: linkedin.com/in/jeremiegirault

VSRE compliant - vsre.info


On 24 août 2015 at 11:59:32, Jérémie Girault (jeremie.girault@snips.net) wrote:

At the moment, pure swift implementation.
We thought about a great system on friday to handle geometrical issues, maybe you will be able to help for the geometric cache implementation. 


-- 
Jérémie Girault
Software Engineer - snips.net
jeremie.girault@snips.net
Mobile: +33 6 81 71 18 16
Skype: jeremie.girault
Twitter: @kamidude
LinkedIn: linkedin.com/in/jeremiegirault

VSRE compliant - vsre.info


On 24 août 2015 at 11:49:53, Mael (notifications@github.com) wrote:

BTW are you storing this in database or using NSCache? also if you need help with all geometrical methods (search locations in a radius, compute distance to location) just ask :)
